crates/rattler_conda_types/src/minimal_prefix_record.rs

```diff
@@ -0,0 +1,936 @@
+        // In the future, we could make Transaction work directly with SparsePrefixRecord.
+        Ok(minimal_records
+            .into_iter()
+            .map(|minimal| {
+                let package_record = minimal.to_package_record();
+                let file_name = format!("{}-{}-{}.tar.bz2",
+                    minimal.name.as_normalized(),
+                    minimal.version,
+                    minimal.build);
+                #[allow(deprecated)]
+                PrefixRecord {
+                    repodata_record: crate::RepoDataRecord {
+                        package_record,
+                        file_name,
+                        url: url::Url::parse("https://conda.anaconda.org/conda-forge/noarch/placeholder-1.0.0-0.tar.bz2").unwrap(),
+                        channel: Some(String::new()),
+                    },
+                    package_tarball_full_path: None,
+                    extracted_package_dir: None,
+                    files: Vec::new(),
+                    paths_data: crate::prefix_record::PrefixPaths::default(),
+                    requested_spec: minimal.requested_spec.clone(),
+                    requested_specs: minimal.requested_specs.clone(),
+                    link: None,
+                    installed_system_menus: Vec::new(),
+                }
+            })
+            .collect())
+    }
+}
+
+/// Struct to hold the parsed fields from JSON
+#[derive(Debug, Default)]
+struct ParsedFields {
+    sha256: Option<Sha256Hash>,
+    md5: Option<Md5Hash>,
+    size: Option<u64>,
+    python_site_packages_path: Option<String>,
+    requested_specs: Vec<String>,
+    requested_spec: Option<String>,
+}
+
+/// Parse minimal JSON fields using nom, stopping at "files" field
+fn parse_minimal_json(input: &str) -> Result<ParsedFields, nom::Err<nom::error::Error<&str>>> {
+    let (_, parsed) = parse_json_object(input)?;
+    Ok(parsed)
+}
+
+/// Parse JSON object looking for specific fields
+fn parse_json_object(input: &str) -> IResult<&str, ParsedFields> {
+    let mut fields = ParsedFields::default();
+
+    let (input, _) = preceded(multispace0, char('{')).parse(input)?;
+    let (input, _) = multispace0(input)?;
+
+    let (remaining, _) = parse_fields(input, &mut fields)?;
+
+    Ok((remaining, fields))
+}
+
+/// Parse JSON fields until we find "files" or reach end
+fn parse_fields<'a>(mut input: &'a str, fields: &mut ParsedFields) -> IResult<&'a str, ()> {
+    loop {
+        // Skip whitespace
+        let (rest, _) = multispace0(input)?;
+        input = rest;
+
+        // Check for end of object
+        if input.starts_with('}') {
+            return Ok((input, ()));
+        }
+
+        // Parse field name
+        let (rest, field_name) = parse_json_string(input)?;
+        input = rest;
+
+        // Check if we hit the "files" field - stop parsing here
+        if field_name == "files" {
```

Comment:
What happens if we are parsing an "old" record where the ordering is not changed? I think we should also cover that case. 

Perhaps checking if we already have the records we need would be safer.  I think the only optional field is the requested-specs field, perhaps we can just emit none or an empty array to ensure the field always exists?

Reply:
That's the main question that I have. Now, if we parse "old" record, it should just put default values to `requested_specs` and `requested_spec` if it can't find them before `"files"`.

What we could do is to don't skip serialization of `requested_specs`. Then if we encounter it before `"files"` then we're sure that we have "new" record and so can stop parsing. If we don't see it, then we would have to parse rest of the file, which should still be faster than the current file. What do you think?